### PR TITLE
Add server-side filter

### PR DESCRIPTION
**Pull request by haosong**

### CODE REVIEW

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+  private final ConcurrentMap<MetricName, AtomicInteger> gauges = new ConcurrentHashMap<>();
+
+  @Nullable
+  private final Tracer tracer;
+
+  @Context
+  private ResourceInfo resourceInfo;
+
+  private WavefrontJaxrsServerFilter(SdkReporter wfJaxrsReporter, ApplicationTags applicationTags,
+                               @Nullable Tracer tracer) {
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java]: not aligned

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+          decorateResponse(containerResponseContext, scope.span());
+          scope.close();
+          scope.span().finish();
+        }
+      } catch (ClassCastException ex) {
+        // no valid scope found
+      }
+    }
+    if (containerRequestContext != null) {
+
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java]: Nit: reduce blank lines where ever possible, to save screen real estate.

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+      if (applicationTags.getCluster() != null) {
+        wfJaxrsReporter.incrementDeltaCounter(new MetricName(responseMetricKey +
+            ".aggregated_per_cluster", aggregatedPerClusterMap));
+      }
+      wfJaxrsReporter.incrementDeltaCounter(new MetricName(responseMetricKey +
+          ".aggregated_per_application", aggregatedPerApplicationMap));
+
+      /*
+       * Overall error response metrics
+       * 1) <prefix>.response.errors.aggregated_per_source (Counter)
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java]: Nit: update the comments to jaxrs.server similar to other comments

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+       * 2) jaxrs.server.response.api.v2.alert.summary.GET.200.cpu_ns
+       */
+      long cpuNanos = ManagementFactory.getThreadMXBean().getCurrentThreadCpuTime() -
+          startTimeCpuNanos.get();
+      wfJaxrsReporter.updateHistogram(new MetricName(responseMetricKey + ".cpu_ns",
+          completeTagsMap), cpuNanos);
+
+      long apiLatency = System.currentTimeMillis() - startTime.get();
+      wfJaxrsReporter.updateHistogram(new MetricName(responseMetricKey + ".latency",
+          completeTagsMap), apiLatency);
```
Comment (srujann) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsServerFilter.java:369]:
```suggestion
          completeTagsMap), apiLatency);
          
      /**
       * total time spent counter: jaxrs.server.response.api.v2.alert.summary.GET.200.total_time
       */
      wfJaxrsReporter.incrementCounter(new MetricName(responseMetricKey + ".total_time",
          completeTagsMap), apiLatency);
```

**Review by srujann — CHANGES_REQUESTED** [review]
There is one missed metric (total_time) and some nits.

**Review by srujann — APPROVED** [review]
LGTM

File: src/test/java/com/wavefront/sdk/jaxrs/JaxrsAppReporterTest.java [hunk 5]
```diff
@@ -0,0 +1,730 @@
...
+
+  @Before
+  public void setup() throws Exception {
+    sampleApp.run("server");
+    httpPort = sampleApp.getHttpPort();
+  }
+
+  @Test
+  public void testCRUD() throws URISyntaxException, IOException {
+//    testCreate();
```
Comment (sushantdewan123) [inline src/test/java/com/wavefront/sdk/jaxrs/JaxrsAppReporterTest.java]: Why are these tests commented? by mistake?

**Review by sushantdewan123 — CHANGES_REQUESTED** [review]

**Review by sushantdewan123 — APPROVED** [review]
LGTM

**Pull request merged**